server/logs/lib.lua

```diff
@@ -0,0 +1,86 @@
+Logs = {}
+Logs.apiUrl= "https://api.fivemanagee.com/logs"
+Logs.apiKey = ""
+
+function Logs:start()
+    Logs.apiKey = GetConvar("FIVEMANAGE_API_KEY", "")
+end
+
+-- TODO: Add context to logs
+function Logs.findContext()
+	local info = debug.getinfo(2, "n")
+
+    return info
+end
+
+function Logs:sendHttpRequest(data)
+	PerformHttpRequest(
+        self.apiUrl,
+        function() end,
+        "POST", 
+        data = json.encode(data),
+        {
+            Authorization = self.apiKey
+        }
+    )
+end
+
+function Logs:LogMessage(info, message, metadata)
+	self.findContext()
+	print("LogMessage: " .. info .. " " .. message)
+
+    local log_data = {
+        info = info,
+        message = message,
+        -- TODO: We might not actually have to do this at all. One seralization might be enough
+        metadata = json.encode(metadata)
+    }
+
+	self:sendHttpRequest(log_data)
+end
+
+function Logs:LogInfo(message, metadata)
+	self.findContext()
+	print("LogMessage: " .. info .. " " .. message)
+
+    local log_data = {
+        level = "info",
+        message = message,
+        -- TODO: We might not actually have to do this at all. One seralization might be enough
+        metadata = json.encode(metadata)
+    }
+
+	self:sendHttpRequest(log_data)
+end
+
+function Logs:LogWarn(message, metadata)
+	self.findContext()
+
+    local log_data = {
+        level = "warn",
+        message = message,
+        -- TODO: We might not actually have to do this at all. One seralization might be enough
+        metadata = json.encode(metadata)
+    }
+
+	self:sendHttpRequest(log_data)
+end
+
+function Logs:LogError(message, metadata)
+	self.findContext()
+
+    local log_data = {
+        level= "error",
+        message = message,
+        -- TODO: We might not actually have to do this at all. One seralization might be enough
+        metadata = json.encode(metadata)
+    }
+
+	self:sendHttpRequest(log_data)
+end
+
```

Comment:
I'm not a fan of the naming schema and syntax like LogError, LogInfo, etc... here and the exports.

I think "Logs" would be better renamed to "Logger" and maybe instead of exposing each method we could allow them to grab the Logger instance something like "exports.fivemanage:getLoggerInstance()" and if we really want to we can just rename the "SendMessage" method and export to "sendLog" if they want to do something manually?

What are your thoughts on that approach?

Reply:
Yeah, we can just create one export for one that is LogMessage and remove the others. Then we can support some getLoggerInstance later. is that ok?